transport/tlscommon/tls.go

```diff
@@ -113,29 +117,24 @@ func ReadPEMFile(log *logp.Logger, s, passphrase string) ([]byte, error) {
 			break
 		}
 
-		if x509.IsEncryptedPEMBlock(block) { //nolint: staticcheck // deprecated, we have to get rid of it
-			var buffer []byte
-			var err error
-			if len(pass) == 0 {
-				err = errors.New("no passphrase available")
-			} else {
-				// Note, decrypting pem might succeed even with wrong password, but
-				// only noise will be stored in buffer in this case.
-				buffer, err = x509.DecryptPEMBlock(block, pass) //nolint: staticcheck // deprecated, we have to get rid of it
+		switch {
+		case x509.IsEncryptedPEMBlock(block): //nolint: staticcheck // deprecated, we have to get rid of it
+			block, err := decryptPKCS1Key(*block, pass)
+			if err != nil {
+				log.Errorf("Dropping encrypted pem with private key '%v': %v", block.Type, err)
+				continue
 			}
-
+			blocks = append(blocks, &block)
+		case block.Type == "ENCRYPTED PRIVATE KEY":
+			block, err := decryptPKCS8Key(*block, pass)
 			if err != nil {
-				log.Errorf("Dropping encrypted pem '%v' block read from %v. %+v",
-					block.Type, r, err)
+				log.Errorf("Dropping encrypted private key '%v': %v", block.Type, err)
```

Comment:
Same thing here, but also adding some context about what was happening:
```suggestion
				log.Errorf("Dropping encrypted pem block with private key, block type '%s', could not decypt as PKCS8: %s", block.Type, err)
```